src/engine/sparqlExpressions/RegexExpression.cpp

```diff
@@ -185,7 +185,9 @@ ExpressionResult RegexExpression::evaluatePrefixRegex(
   lowerAndUpperIds.reserve(actualPrefixes.size());
   for (const auto& prefix : actualPrefixes) {
     lowerAndUpperIds.emplace_back(
-        context->_qec.getIndex().prefix_range(prefix));
+        context->_qec.getIndex().prefix_range_internal_vocabulary(prefix));
+    lowerAndUpperIds.emplace_back(
+        context->_qec.getIndex().prefix_range_external_vocabulary(prefix));
```

Comment:
Should be a single function that returns ALL the ranges at once, to make this more agnostic to changes of the vocabulary structure in the future.